### DISCUSSION TITLE
GUI Apps not starting: MESA: error: ZINK: failed to choose pdev

### DISCUSSION THREAD

**Discussion by chris-neub** [microsoft/WSL#12243] [Q&A]
Hi,

Infos:
```
WSL-Version: 2.3.24.0
Kernelversion: 5.15.153.1-2
WSLg-Version: 1.0.65
MSRDC-Version: 1.2.5620
Direct3D-Version: 1.611.1-81528511
DXCore-Version: 10.0.26100.1-240331-1435.ge-release
Windows-Version: 10.0.22631.4317

WSLg ( x86_64 ): 1.0.65+Branch.main.Sha.25d8697d5049e1994baa6ba542f507347b608ad4
Built at: Wed Aug 14 21:11:14 UTC 2024
Mariner: VERSION="2.0.20240609"
DirectX-Headers:
mesa:
pulseaudio: 6f045ff0dca233a939a2aba815f84d177e294122
FreeRDP: c4030980b29322a9cb2190711a5fadeeeb8b6a33
weston: f227edd681479ec3cb2290a25d84d2d3462aebfa

drwxrwxrwx  2 root      root        60 Nov  6 12:05 .
drwxrwxrwt 14 root      root      4096 Nov  6 12:06 ..
srwxrwxrwx  1 mindberry mindberry    0 Nov  6 12:05 X0
```

I'm very new to WSL and not a very good Linux User, but want to start with it. I've installed WSL according to windows learn and updated and upgraded it. However when I want to open an GUI like nautilus I always get this errors and it will open after a while:

```
** Message: 12:01:54.893: Connecting to org.freedesktop.Tracker3.Miner.Files
MESA: error: ZINK: failed to choose pdev
libEGL warning: egl: failed to create dri2 screen
MESA: error: ZINK: failed to choose pdev
glx: failed to create drisw screen
```

I've tried also to change the graphics adapter from Intel to Nvidia - but no difference. Drivers should be installed properly. I also have the same issues with an AMD notebook. So what do I miss?

**Comment by GitLuci** [comment]
if u find the answer, please share with usss :)))

**Comment by GitLuci** [comment]
I find this one, can be a way

https://askubuntu.com/questions/1530856/wsl2-mesa-error-zink-failed-to-choose-pdev-after-glxinfo-b

**Reply by chris-neub** [reply]
Hi, thanks for the link. I've also found it und unfortunately it didn't help me with my problem. 
I've also read something that this is a known bug, but haven't found much about it.

**Comment by Rhyzhang** [comment]
Hi!

I'm also having this problem so I opened an issue hoping it gets viability: https://github.com/microsoft/WSL/issues/12362

**Comment by tom-huntington** [comment]
[478332](https://github.com/microsoft/wslg/issues/1250#issuecomment-2646132024) worked for me
